app/main.py

```diff
@@ -1,6 +1,20 @@
-from typing import Callable
+from typing import Callable, Any, Tuple
 
 
-def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+def cache(func: Callable[..., Any]) -> Callable[..., Any]:
+    results_cache = {}
+
+    def wrapper(*args: Tuple, **kwargs: Any) -> Any:
```

Comment:
kwargs is a dictionary, and you don't really need to specify types for args and kwargs here